rosetta/invoker/invoker.go

```diff
@@ -36,21 +38,44 @@ import (
 type Invoker struct {
 	index index.Reader
 	vm    *fvm.VirtualMachine
-	reads map[uint64]delta.GetRegisterFunc
+	cache *ristretto.Cache
 }
 
-func New(index index.Reader) *Invoker {
+func New(index index.Reader, options ...func(*Config)) (*Invoker, error) {
 
+	// Initialize the invoker configuration with conservative default values.
+	cfg := Config{
+		CacheSize: uint64(100 * datasize.MB), // ~100 MB default size
+	}
+
+	// Apply the option parameters provided by consumer.
+	for _, option := range options {
+		option(&cfg)
+	}
+
+	// Initialize interpreter and virtual machine for execution.
 	rt := fvm.NewInterpreterRuntime()
 	vm := fvm.NewVirtualMachine(rt)
 
+	// Initialize the Ristretto cache with the size limit. Ristretto recommends
+	// keeping ten times as many counters as items in the cache when full.
+	// Assuming an average item size of 1 kilobyte, this is what we get.
+	cache, err := ristretto.NewCache(&ristretto.Config{
+		NumCounters: int64(cfg.CacheSize) / 1000 * 10,
```

Comment:
```suggestion
		NumCounters: int64(cfg.CacheSize) / 1024 * 10,
```

Nitpick

Reply:
The Ristretto team (which is the Badger team, which is the Dgraph team) uses `1000` as the factor across their code bases, so I'm just staying consistent with them.